libs/platform/ebpf_epoch.c

```diff
@@ -66,11 +66,11 @@ typedef struct _ebpf_epoch_state
 typedef struct _ebpf_epoch_cpu_entry
 {
     ebpf_lock_t lock;
-    _Requires_lock_held_(lock) ebpf_epoch_state_t epoch_state;                 // Per-CPU epoch state.
-    _Requires_lock_held_(lock) ebpf_list_entry_t free_list;                    // Per-CPU free list.
-    _Requires_lock_held_(lock) ebpf_hash_table_t* thread_table;                // Per-CPU thread table.
-    _Requires_lock_held_(lock) ebpf_non_preemptible_work_item_t* stale_worker; // Per-CPU stale worker DPC.
-    uint32_t padding; // Pad to multiple of EBPF_CACHE_LINE_SIZE.
+    _Guarded_by_(lock) ebpf_epoch_state_t epoch_state;                 // Per-CPU epoch state.
```

Comment:
Question: So you use `_Requires_lock_held_` for functions but `_Guarded_by_` for types/variables?

Reply:
Yeah, I found this out the hard way. Setting _Requires_lock_held_ on type doesn't raise a warning, but also doesn't actually trigger a code analysis warning for cased where the lock isn't held.

Reply:
https://learn.microsoft.com/en-us/cpp/code-quality/annotating-locking-behavior?view=msvc-170